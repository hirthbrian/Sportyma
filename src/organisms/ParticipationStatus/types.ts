export interface Props {
  eventId: number;
}
