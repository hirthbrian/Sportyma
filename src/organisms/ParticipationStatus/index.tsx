const ParticipationStatus = () => {
  return <></>;
};

export default ParticipationStatus;
