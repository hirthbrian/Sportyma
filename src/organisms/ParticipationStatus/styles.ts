import styled from "styled-components/native";

export const Container = styled.View`
  padding-top: 15px;
  flex-direction: row;
`;

export const HorizontalSeparator = styled.View`
  width: 15px;
`;
