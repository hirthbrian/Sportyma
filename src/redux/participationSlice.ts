import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import type { RootState } from "./store";

import trainig1 from "../../data/training_1.json";
import trainig2 from "../../data/training_2.json";

// Define a type for the slice state
interface ParticipationState {
  [key: number]: string | null;
}

// Define the initial state using that type
const initialState: ParticipationState = {
  1: trainig1.participation,
  2: trainig2.participation,
};

export const participationSlice = createSlice({
  name: "participation",
  // `createSlice` will infer the state type from the `initialState` argument
  initialState,
  reducers: {
    approve: (state, action: PayloadAction<number>) => {
      state[action.payload] = "PRESENT";
    },
    refuse: (state, action: PayloadAction<number>) => {
      state[action.payload] = "ABSENT";
    },
  },
});

export const { approve, refuse } = participationSlice.actions;

// Other code such as selectors can use the imported `RootState` type
export const selectParticipationStatus = (state: RootState) =>
  state.participation;

export default participationSlice.reducer;
