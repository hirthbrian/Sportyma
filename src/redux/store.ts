import { configureStore } from "@reduxjs/toolkit";
import participation from "./participationSlice";

export const store = configureStore({
  reducer: {
    participation,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
