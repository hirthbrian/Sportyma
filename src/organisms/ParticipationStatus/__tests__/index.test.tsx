import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

import { store } from "../../../redux/store";

import ParticipationStatus from "..";

describe("<ParticipationStatus />", () => {
  let props;

  beforeEach(() => {
    props = {
      eventId: 1,
    };
  });

  it("renders correcly", () => {
    expect(
      renderer
        .create(
          <Provider store={store}>
            <ParticipationStatus {...props} />
          </Provider>
        )
        .toJSON()
    ).toBeTruthy();
  });
  it("match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ParticipationStatus {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
