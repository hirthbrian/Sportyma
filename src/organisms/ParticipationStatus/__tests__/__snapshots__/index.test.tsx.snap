// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipationStatus /> match snapshot 1`] = `
Array [
  <Text
    size={20}
    style={
      Array [
        Object {
          "color": "#152D4B",
          "fontFamily": "Kanit-Medium",
          "fontSize": 20,
        },
      ]
    }
    weight={500}
  >
    Ma participation
  </Text>,
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingTop": 15,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          color="#29D48C"
          isActive={false}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#2559f233",
                "borderRadius": 50,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            color="#29D48C"
            isActive={false}
            source={1}
            style={
              Array [
                Object {
                  "height": 18,
                  "marginRight": 10,
                  "tintColor": "#2559F2",
                  "width": 18,
                },
              ]
            }
          />
          <Text
            isActive={false}
            size={13}
            style={
              Array [
                Object {
                  "color": "#2559F2",
                  "fontFamily": "Kanit-Medium",
                  "fontSize": 13,
                  "letterSpacing": 1,
                },
              ]
            }
            weight={500}
          >
            PRÃ‰SENT
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "width": 15,
          },
        ]
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          color="#EF4948"
          isActive={false}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#2559f233",
                "borderRadius": 50,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            color="#EF4948"
            isActive={false}
            source={1}
            style={
              Array [
                Object {
                  "height": 18,
                  "marginRight": 10,
                  "tintColor": "#2559F2",
                  "width": 18,
                },
              ]
            }
          />
          <Text
            isActive={false}
            size={13}
            style={
              Array [
                Object {
                  "color": "#2559F2",
                  "fontFamily": "Kanit-Medium",
                  "fontSize": 13,
                  "letterSpacing": 1,
                },
              ]
            }
            weight={500}
          >
            ABSENT
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
